module4/labs/Week3/MaintainingState.md

```diff
@@ -0,0 +1,48 @@
+---
+layout: page
+title: Maintaining State Lab
+---
+
+In today's lab, we are going to build on the [Start Repo](https://github.com/turingschool-examples/GoodBooksMvcStarter) from the lesson.
+
+<section class='instructor-notes' markdown='1'>
+
+Before letting students tackle this, go over the user stories as a group.  discuss what favoriting is, and some options for showing if a book has been favorited.  Let students know that they may need to use the following (refresh their memories on):
+* String manipulation
+* Working with collections
+* html/css
+
+</section>
+
+### Favoriting Books
```

Comment:
I like the challenge level of this lab. I think it requires them to utilize a lot of different concepts. I wonder if it's a good candidate for some pair programming with guidance from the instructor about when to switch?

Also might be too nuanced, but it feels a bit off for students to leave this lab thinking that cookies are the way to implement favoriting, because in real world you would almost certainly want to persist favorites to the database somehow. Maybe a discussion at the end of the lab about pros and cons of using cookies to track favorites would be useful?

I'm also thinking of how to introduce something to this lab that is a very real world case of using a cookie. What if there is a spicy part two at the end of the lab where they use a cookie for consent management. Like the first time the user visits the page they get a message saying something like "This website uses cookies to ensure you get the best experience on our website." and buttons to consent or not consent" And then then set a cookie of if they consent or not and only display the like like buttons if they consent?

Maybe that's too spicy? Not sure.